stac_generator/base/generator.py

```diff
@@ -1,83 +1,113 @@
+from __future__ import annotations
```

Comment:
Is this really necessary?

Reply:
Not in this particular scenario. I must have had a `TYPE_CHECKING` block that got removed. Having it there doesn't hurt. In fact there were some [considerations](https://docs.python.org/3/library/__future__.html) (see footnote) for having this by default 

It's useful because you can do two things: 

- Annotate names that get `forwardRefef` later: in regular python, class names don't get registered to the local namespace until its definition is complete. The consequence of this is the following is illegal since the name `Node` is undefined.

```python
class Node: 
  def __init__(self, val: int, next_node: Node)->None: 
    self.val = val 
    self.next_node = node 
```

The canonical way of avoiding this is by annotating Node as a string: 

```python
class Node: 
  def __init__(self, val: int, next_node: "Node")->None: 
    self.val = val 
    self.next_node = node 
```
This tells static analysis tools that `Node` is a name that will get resolved in the `future` - i.e. once definition is complete (the technical term is forward referencing). This looks ugly so people invent `from __future__ import annotations` which essentially do the same thing. 

- Allows for `if TYPE_CHECKING:` import optimisation. Importing some modules to import types is bad for speed, so python invents the `if TYPE_CHECKING` syntax that only import the type annotation modules at static analysis instead of run time.

Reply:
All that is well and good but I would simply not use anything from future in a production system. Especially in this case where the release for this functionality has been delayed indefinitely. Simply use the string type hints even if its ugly.